services/offline/src/lib/cached-sections.js

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types'
+import React, { createContext, useContext, useState, useEffect } from 'react'
+import { useOfflineInterface } from './offline-interface'
+
+const CachedSectionsContext = createContext()
```

Comment:
This sets up context for the status of cached sections. It will be refactored to decrease rerenders in consuming components.

Copied from above:
Question: `useCachedSections` and `useCachedSection` are not currently exported from the package because the `useCachedSection` functionality is re-returned from `useCacheableSection`. Getting a list of cached sections' statuses may be useful for an app though, for example when purging items from the cache that don't exist on the server any more. What do you think?